lib/babe/verify_test.go

```diff
@@ -52,9 +52,9 @@ func newEncodedBabeDigest(t *testing.T, value scale.VaryingDataTypeValue) []byte
 	return enc
 }
 
-func encodeAndHashHeader(t *testing.T, header *types.Header) common.Hash {
+func encodeAndHashHeader(t *testing.T, header types.Header) common.Hash {
```

Comment:
- I didn't realise it was that big, my bad! (It's 184 bytes by the way)
- `to avoid heap growth by copying the struct value` it won't be on the heap but on the stack if there is no pointer to it, which is faster and puts less GC pressure

Anyway, I ran a little benchmark comparing passing by value and by pointer that header and:

```
Benchmark_Header_ValueVsPointer/value-24         	417870074	         2.880 ns/op	       0 B/op	       0 allocs/op
Benchmark_Header_ValueVsPointer/pointer-24       	1000000000	         0.4167 ns/op	       0 B/op	       0 allocs/op
```

So pointer it is!

I also created https://github.com/ChainSafe/gossamer/issues/2813 as an enhancement for the `types.Block` structure which is used as a value for now.